src/ci/docker/host-x86_64/x86_64-gnu-llvm-13/Dockerfile

```diff
@@ -44,7 +44,7 @@ ENV NO_DOWNLOAD_CI_LLVM 1
 # Using llvm-link-shared due to libffi issues -- see #34486
 ENV RUST_CONFIGURE_ARGS \
       --build=x86_64-unknown-linux-gnu \
-      --llvm-root=/usr/lib/llvm-13 \
+      --llvm-root=/usr/lib/llvm-14 \
```

Comment:
These changes are very weird, considering that the file is named `src/ci/docker/host-x86_64/x86_64-gnu-llvm-`**`13`**`/Dockerfile`

Reply:
This is a temporary hack to test in CI if it works. If we do want to test LLVM 14 this should probably go in a new runner.

Maybe we can have two new runners for building rust with preinstalled LLVM 14 and 15 and that run in CI if and only if `tests/assembly` or `tests/codegen` is changed?

I didn't actually intend to change CI with this PR.

Reply:
It's probably time for us to drop LLVM 13 altogether, at which point I usually just rename the current minimum runner. I've also just asked on Zulip `#t-infra` about capacity for adding more runners. But either way, you're right not to mix that in this PR.

Reply:
cc https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/testing.20multiple.20llvm.20versions